### ISSUE TITLE
Multiple plugins are having JSON decoding errors

### ISSUE THREAD

**Issue by Techman** [snoonetIRC/CloudBot#150]
Distro Information:
```
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.3 LTS
Release:        16.04
Codename:       xenial

Python 3.5.2
pip 9.0.1 from /usr/local/lib/python3.5/dist-packages (python 3.5)
```
---------------------------
```
[01:21:35] [INFO] [techtronix:#lounge] <Techman> .wa 5+5
[01:21:36] [ERROR] Error in hook wolframalpha:wolframalpha
Traceback (most recent call last):
  File "/home/ircd/Cloudbot/cloudbot/plugin.py", line 483, in internal_launch
    out = yield from task
  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/ircd/Cloudbot/cloudbot/plugin.py", line 446, in _execute_hook_threaded
    return hook.function(*parameters)
  File "/home/ircd/Cloudbot/plugins/wolframalpha.py", line 43, in wolframalpha
    short_url = web.try_shorten(query_url.format(urllib.parse.quote_plus(text)))
  File "/home/ircd/Cloudbot/cloudbot/util/web.py", line 67, in try_shorten
    return impl.try_shorten(url, custom, key)
  File "/home/ircd/Cloudbot/cloudbot/util/web.py", line 109, in try_shorten
    return self.shorten(url, custom, key)
  File "/home/ircd/Cloudbot/cloudbot/util/web.py", line 154, in shorten
    j = r.json()
  File "/usr/local/lib/python3.5/dist-packages/requests/models.py", line 892, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.5/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
[01:21:42] [INFO] [techtronix:#lounge] <Techman> nope, still errors
```

**Comment by Techman** [comment]
Looks like it is not just WolframAlpha.
```
[01:32:44] [INFO] [techtronix:#lounge] <Techman> @np
[01:32:44] [ERROR] Error in hook lastfm:lastfm
Traceback (most recent call last):
  File "/home/ircd/Cloudbot/cloudbot/plugin.py", line 483, in internal_launch
    out = yield from task
  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/ircd/Cloudbot/cloudbot/plugin.py", line 446, in _execute_hook_threaded
    return hook.function(*parameters)
  File "/home/ircd/Cloudbot/plugins/lastfm.py", line 239, in lastfm
    url = web.try_shorten(track["url"])
  File "/home/ircd/Cloudbot/cloudbot/util/web.py", line 67, in try_shorten
    return impl.try_shorten(url, custom, key)
  File "/home/ircd/Cloudbot/cloudbot/util/web.py", line 109, in try_shorten
    return self.shorten(url, custom, key)
  File "/home/ircd/Cloudbot/cloudbot/util/web.py", line 154, in shorten
    j = r.json()
  File "/usr/local/lib/python3.5/dist-packages/requests/models.py", line 892, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3.5/json/__init__.py", line 319, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.5/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.5/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

**Comment by linuxdaemon** [comment]
If you look at the stack trace, the issue isn't in a specific hook, it's occurring in `web.try_shorten()`. I have a [branch](https://github.com/linuxdaemon/CloudBot/tree/gonzobot%2Bweb-shorten-raise) which will raise status errors, because the issue is most likely that is.gd is returning some error

**Comment by linuxdaemon** [comment]
I believe this would have been fixed in #189 so I'm going to go ahead and close it.

**Comment by Techman** [comment]
Seems to be fixed, indeed.
